### PR TITLE
chore: migrate Popover

### PR DESCRIPTION
**Pull request by RobelTekle**
## DESCRIPTION
migrate Popover component.
## BREAKING CHANGES:
- The `usePopoverHover` and `usePopover` no longer accept the `animated` prop. This was deprecated by Ariakit. But `hideTimeout` and `showTimeout` props are still accepted. 

## SCREENSHOTS / SCREEN RECORDINGS

https://github.com/user-attachments/assets/b95bf1b0-23f9-4b85-9f2e-49c471b2a6fb

## COMPATIBILITY

- [ ] Tested on Safari (desktop)
- [ ] Tested on Chrome (desktop)
- [ ] Tested on Firefox (desktop)
- [ ] Tested on mobile device sizes
- [ ] Tested on tablet device sizes
- [ ] Tested on IOS Safari (either device or simulator)

## QA

- [ ] Thoroughly tested in local environment
- [ ] Added tests for all new features
- [ ] Added tests that considered edge cases

### CODE REVIEW

File: lib/src/components/Popover/Arrow.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  const currentPlacement = Ariakit.useStoreState(store, 'currentPlacement')
+
+  const [parentPlacement] = currentPlacement.split('-')
+  const transform = rotation[parentPlacement as keyof typeof rotation]
+
+  return (
+    <Ariakit.PopoverArrow className={cx('arrow')} store={store}>
+      <svg
+        className={cx('arrow-item')}
+        style={{ '--transform': transform } as React.CSSProperties & Record<string, string>}
```
Comment (anucreative) [inline lib/src/components/Popover/Arrow.tsx]: Do we need the `Record<string, string>` ?

File: lib/src/components/Popover/Content.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { forwardRef } from 'react'
+
+import { classNames } from '@/utils'
+
+import popoverStyles from './popover.module.scss'
+const cx = classNames(popoverStyles)
+
+export const Content = forwardRef<HTMLDivElement, React.HTMLProps<HTMLDivElement>>((props, ref) => {
```
Comment (anucreative) [inline lib/src/components/Popover/Content.tsx]: I think we want to be able to pass Tailwind classes to the `Content` component, non ?

File: lib/src/components/Popover/popover.module.scss [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    top: 1px;
+    right: 1px;
+    margin-left: var(--spacing-md);
+  }
+
+  .arrow {
+    color: var(--color-neutral-90);
+  }
+  .arrow-item {
+    transform: var(--transform);
```
Comment (anucreative) [inline lib/src/components/Popover/popover.module.scss]: Where does `--transform` come from ?

Comment (anucreative) [reply lib/src/components/Popover/popover.module.scss]: Ah, from `Popover.tsx`. If we rename it to `--popover-transform` would that make it more obvious ?

File: lib/src/components/Popover/Popover.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      </Ariakit.Popover>
+    )
+  }
+)
+
+export const Popover = Object.assign(PopoverComponent, {
+  Content,
+  Title,
+  Trigger: PopoverTrigger,
+})
```
Comment (anucreative) [inline lib/src/components/Popover/Popover.tsx]: We already have a barrel file `Popover/index.tsx‎`. I would put everything together in one place (either the barrel file or in this file).

Whichever pattern we use (barrel file or export from main file) we should use it across all components.

File: lib/src/components/Popover/Trigger.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+import type { UsePopover, UsePopoverHover } from './usePopover'
+
+export type PopoverTriggerProps = React.ButtonHTMLAttributes<HTMLButtonElement> & {
+  store: UsePopover
+}
+
+export const PopoverTrigger = forwardRefWithAs<PopoverTriggerProps, 'button'>(
+  ({ as: As, store, ...rest }, ref) => {
+    return (
+      <Ariakit.PopoverDisclosure
```
Comment (anucreative) [inline lib/src/components/Popover/Trigger.tsx:14]: We want to be able to override with Tailwind, non ?

File: lib/src/components/Popover/Trigger.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+      />
+    )
+  }
+)
+
+export type PopoverHoverTriggerProps = React.HTMLAttributes<HTMLAnchorElement> & {
+  store: UsePopoverHover
+}
+
+export const PopoverHoverTrigger = forwardRefWithAs<PopoverHoverTriggerProps, 'a'>(
```
Comment (anucreative) [inline lib/src/components/Popover/Trigger.tsx:29]: We want to be able to override with Tailwind, non ?

File: website/build-app/pages/foundations/upgrades/v9.mdx [hunk 7]
```diff
@@ -46,3 +46,12 @@ Size props are only for intern values (xs, sm, ...), you can now use tailwind si
 -<CheckIcon size="24"  />
 +<CheckIcon className="size-[24px]" />
```
Comment (anucreative) [inline website/build-app/pages/foundations/upgrades/v9.mdx]:
```suggestion
+<CheckIcon className="size-(--spacing-xl)" />
```
